bids/tests/test_visualize.py

```diff
@@ -0,0 +1,153 @@
+from pathlib import Path
+
+import shutil
+
+import pytest
+
+from bids.layout import BIDSLayout
+from bids.visualize import EventPlotter  # type: ignore
+from bids.visualize import LayoutPlotter  # type: ignore
+from bids.tests import get_test_data_path
+
+"""TODO
+-   test on datasets with a large range of events types
+"""
+
+
+@pytest.mark.parametrize(
+    "output_dir",
+    [
+        (Path(__file__).parent.joinpath("tmp")),
+        (None),
+    ],
+)
+@pytest.mark.parametrize(
+    "show",
+    [
+        (True),
+        (False),
+    ],
+)
+@pytest.mark.parametrize(
+    "dataset,filters,plot_by",
+    [
+        ("ds114", None, "suffix"),
+        ("eeg_ds003654s_hed", None, ["suffix", "run"]),
+        ("ds000117", dict(session=["mri", "meg"]), None),
+        ("fnirs_tapping", None, "suffix"),
+    ],
+)
+def test_LayoutPlotter_smoke(
+    dataset, filters, plot_by, show, output_dir, bids_examples
+):
+
+    layout = BIDSLayout(root=Path(bids_examples).joinpath(dataset))
+
+    LayoutPlotter(layout, filters=filters).plot(
+        plot_by=plot_by, show=show, output_dir=output_dir
+    )
+    if output_dir is not None:
+        shutil.rmtree(output_dir)
+
+
+def test_LayoutPlotter_missing_subject(bids_examples):
+
+    layout = BIDSLayout(root=Path(bids_examples).joinpath("ds114"))
```

Comment:
FYI the idiomatic use of Paths is to use the division operator to extend:

```suggestion
    layout = BIDSLayout(root=Path(bids_examples) / "ds114")
```

Reply:
should have fixed that now